src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx

```diff
@@ -29,6 +49,180 @@ class WidgetCard extends React.Component<Props> {
     return false;
   }
 
+  renderWidgetQuery(widgetQuery: WidgetQuery, index: number) {
+    const {location, router, selection, api, organization, widget} = this.props;
+
+    const statsPeriod = selection.datetime.period;
+    const {start, end} = selection.datetime;
+    const {projects, environments} = selection;
+
+    const datetimeSelection = {
+      start: start || null,
+      end: end || null,
+      period: statsPeriod,
+    };
+
+    const legend = {
+      right: 10,
+      top: 0,
+      icon: 'circle',
+      itemHeight: 8,
+      itemWidth: 8,
+      itemGap: 12,
+      align: 'left',
+      textStyle: {
+        verticalAlign: 'top',
+        fontSize: 11,
+        fontFamily: 'Rubik',
+      },
+      selected: getSeriesSelection(location),
+      formatter: seriesName => {
+        const arg = getAggregateArg(seriesName);
+        if (arg !== null) {
+          const slug = getMeasurementSlug(arg);
+          if (slug !== null) {
+            seriesName = slug.toUpperCase();
+          }
+        }
+        return seriesName;
+      },
+    };
+
+    const chartOptions = {
+      grid: {
+        left: '10px',
+        right: '10px',
+        top: '40px',
+        bottom: '0px',
+      },
+      seriesOptions: {
+        showSymbol: false,
+      },
+      tooltip: {
+        trigger: 'axis',
+        valueFormatter: tooltipFormatter,
+      },
+      yAxis: {
+        axisLabel: {
+          color: theme.chartLabel,
+          formatter: (value: number) =>
+            axisLabelFormatter(value, widgetQuery.fields[0] ?? ''),
+        },
+      },
+    };
+
+    const yAxis = [...widgetQuery.fields];
+
+    return (
+      <ChartZoom
```

Comment:
I feel like we have this chart component tree in a few places.  This one is a little different because of the chart component selection in the middle, but if we extract that out we should be able to refactor this to share code with other charts later.